service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java

```diff
@@ -75,6 +88,9 @@ public final void setTracer(final Tracer opentracingTracer) {
      */
     @Override
     public final void start(final Future<Void> startFuture) {
+        if (healthCheckServer != null) {
```

Comment:
based on our discussions, we should add a log statement here for the `null` case, right?